infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSManager.java

```diff
@@ -0,0 +1,198 @@
+            Deployment deployment = new Deployment();
+            deployment.setHosts(hosts);
+            return deployment;
+        } catch (InterruptedException e) {
+            throw new TestGridInfrastructureException("Error occured while waiting for " +
+                    "CloudFormation Stack creation", e);
+        } catch (IOException e) {
+            throw new TestGridInfrastructureException("Error occured while Reading CloudFormation script", e);
+        }
+
+    }
+
+    /**
+     * This method waits for completion of AWS process and generate result depending on
+     * the result code.
+     *
+     * @param stackBuilder AWS CF builder object.
+     * @param stackName    The stack name to perform the waiting upon.
+     * @return true or false depending on the result.
+     * @throws InterruptedException            when an Interrupt occurs while waiting for CF result.
+     * @throws TestGridInfrastructureException when an error occurs while reading the cf template file.
+     */
+    private boolean waitForAWSProcess(AmazonCloudFormation stackBuilder, String stackName) throws InterruptedException,
+            TestGridInfrastructureException {
+        DescribeStacksRequest wait = new DescribeStacksRequest();
+        wait.setStackName(stackName);
+        boolean completed = false;
+        boolean successful = false;
+        log.info("Waiting ..");
+        while (!completed) {
+            List<Stack> stacks = stackBuilder.describeStacks(wait).getStacks();
+            if (stacks.isEmpty()) {
+                throw new TestGridInfrastructureException("There is no stack for the name :" + stackName);
+            } else {
+                for (Stack stack : stacks) {
+                    if (stack.getStackStatus().equals(StackStatus.CREATE_COMPLETE.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.DELETE_COMPLETE.toString())) {
+                        completed = true;
+                        successful = true;
+                    } else if (stack.getStackStatus().equals(StackStatus.CREATE_FAILED.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_FAILED.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_COMPLETE.toString())) {
+                        completed = true;
+                        successful = false;
+                    } else if (stack.getStackStatus().equals(StackStatus.CREATE_IN_PROGRESS.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.DELETE_IN_PROGRESS.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_IN_PROGRESS.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.REVIEW_IN_PROGRESS.toString())) {
+                        completed = false;
+                    }
+                }
+            }
+            if (!completed) {
+                Thread.sleep(5000);
```

Comment:
Do not keep values without a meaning or add a comment to describe it.

Reply:
Added a comment